Delete employee/on4 t 39 new endpoint

wok-gocaspi opened the pull request:
Verify ON4T-39

wok-gocaspi committed:
added new delete route with handler and database method.
!IMPORTANT! not tested yet

wok-gocaspi committed:
added tests and fixed issue on database
+ added tests for deleteUser Handler
+ added tests for deleteemployeebyid for servicehandler
/ changed mongodb url to match local enviroment instance

nil-gocaspi commented:
If I call the delte endpoint with a query parameter which is not assigned to an employee in the database, the endpoint returns statuscode 200 (ok). It should return an error message and status 400 (badrequest).

wok-gocaspi committed:
Delete Endpoint now returns a error if no user exists
+ DeleteByID now returns two values, one as the default and the other if there have been no user
service and handlers have been modified accordingly
+ added Postman Collection

nil-gocaspi commented:
In the databaseClient.go file:
-  DeleteById returns 2 mongoDeleteResults. It should return   mongoDeleteResult,error
- Please no fmt.Println in the PR
In the registerService.go file:
-  DeleteEmployeeById return 2 mongoDeleteResults. It should return   mongoDeleteResult,error
In the Handler.go file:
-Please no fmt.Println
-tests for DeleteEmployeeHandler do not work, because DeleteEmployeeByIdReturns cannot return string "1"

wok-gocaspi committed:
Database Method DeleteByID got changed
The Method now returns a delete result and error
instead of two delete results

wok-gocaspi committed:
removed fmt prints

wok-gocaspi committed:
removed broken tests

nil-gocaspi commented:
OK!
